charts/drag-provider/src/DragProvider.spec.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {
+  Chart,
+  ChartCard,
+  Grid,
+  Header,
+  Line,
+  SortDirection,
+  SortKey,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from '@lg-charts/core';
+import { fireEvent, Matcher, render, screen } from '@testing-library/react';
+
+import { DragProvider } from './DragProvider';
+
+function renderChart(dragId: string) {
+  return (
+    <Chart key={dragId} dragId={dragId}>
+      <Header title={dragId} showDivider data-testid={`${dragId}-header`} />
+      <Grid vertical={false} />
+      <Tooltip sortDirection={SortDirection.Desc} sortKey={SortKey.Value} />
+      <XAxis type="value" />
+      <YAxis type="value" />
+      <Line
+        name="Line 1"
+        data={[
+          [0, 0],
+          [1, 1],
+          [2, 2],
+          [3, 3],
+        ]}
+      />
+    </Chart>
+  );
+}
+
+async function clickAndDragByTestId(id: Matcher) {
+  const header = screen.getByTestId(id);
+  fireEvent.mouseDown(header, { clientX: 0, clientY: 0 });
+  fireEvent.mouseMove(header, { clientX: 100, clientY: 100 });
+  fireEvent.mouseUp(header);
+}
```

Comment:
Future: might be worth exporting some sort of drag-n-drop event along with test harnesses (since `userEvent` doesn't support `.drag`)